scripts/generate_sitemap.py

```diff
@@ -196,12 +197,32 @@ def generate(self):
                 self.log_errors('NODE', obj['guids___id'], e)
             progress.increment()
         progress.stop()
+        #Removed previous logic as it blocked withdrawn preprints to get in sitemap generator
+        objs = Preprint.objects.filter(
+            Q(is_published=True, date_withdrawn__isnull=True) | Q(date_withdrawn__isnull=False)
+        ).annotate(
+            most_recent_non_withdrawn=Subquery(
+                Preprint.objects.filter(
+                    guids=OuterRef('guids')
+                ).filter(
+                    Q(is_published=True, date_withdrawn__isnull=True)  # Non-withdrawn and published
+                ).order_by('-versioned_guids__version').values('versioned_guids__version')[:1]
+            )
+        ).annotate(
+            most_recent_withdrawn=Subquery(
+                Preprint.objects.filter(
+                    guids=OuterRef('guids')
+                ).filter(
+                    date_withdrawn__isnull=False  # Withdrawn versions
+                ).order_by('-versioned_guids__version').values('versioned_guids__version')[:1]
+            )
+        ).order_by(
+            'versioned_guids__guid_id',
+            '-is_published',
+            '-versioned_guids__version'
+        ).distinct('versioned_guids__guid_id')
```

Comment:
This works and 👍 for using subquery/annotation for aggregation. However, it seems more complicated than we actually need. 

-[ ] Here what I have in mind:

* Use `.filter(date_published__isnull=False)`, which is equivalent to `Q(is_published=True, date_withdrawn__isnull=True) | Q(date_withdrawn__isnull=False)`

* Without the `Q() | Q()`, I think we only need one subquery annotation to aggregate the latest version.

Reply:
Another thing: not in this PR but in the future, the first item can be replaced with the `EverPublishedPreprintManager` in https://github.com/CenterForOpenScience/osf.io/pull/10834.

Reply:
Finally, this doesn't blocking my merge, when you have time, probably when you are doing the unit tests ticket for the sitemap, can you drop the sitemap generated by your local env in the ticket.